http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -461,4 +544,124 @@ private void addScopes(String scopes, Set<String> allScopes) {
             allScopes.addAll(Arrays.asList(scopes.split("\\s+")));
         }
     }
+
+    private String convertToRequestParameter(URIBuilder redirectUriBuilder, String redirectUri, String state, List<NameValuePair> forwardedQueryParams) throws JoseException, IOException {
+        redirectUriBuilder.addParameter(SCOPE, OIDC_SCOPE);
+
+        JwtClaims jwtClaims = new JwtClaims();
+        jwtClaims.setIssuer(deployment.getIssuerUrl());
+        jwtClaims.setAudience(deployment.getProviderUrl());
```

Comment:
From the spec, "the aud value SHOULD be or include the OP's Issuer Identifier URL". So getIssuerUrl should be used here.

Reply:
Looks the iss and aud claims should be included if the request object will be signed.

Reply:
Thanks! I will update the issuer and audience. 
The request object is always signed (in a sense that we call `jsonWebSignature.sign()` function), even if it is signed using NONE algorithm. Unless signing using NONE algorithm doesn't count as a signature. 
If NONE does not count, then I can change it to only add those headers if the signing algorithm is not NONE. 
Btw, NONE is the default value of the signing algorithm.